### PR TITLE
Add Trip object class

### LINKED ISSUE #61: Remove overloaded constructor in Person class
As discussed in PR #57 we could improve code quality by removing the overloaded constructor, and adding classes to handle the optional driver type.

### PR DESCRIPTION
**Pull request by ZechariahTan**
* Modify packages to reflect planned Trip implementation

### CODE REVIEW

File: src/main/java/seedu/address/model/trip/Trip.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * Represents a Trip.
+ * Guarantees: details relevant for a trip are present and not null, field values are validated, immutable.
+ */
+public class Trip {
+
+    // Data fields
+    private final TripDay tripDay;
+    private final TripTime tripTime;
+    private final Set<Passenger> passengers = new HashSet<>();
+    private final Set<Tag> tags = new HashSet<>();
```
Comment (JoelHo) [inline src/main/java/seedu/address/model/trip/Trip.java]: This should be in the constructor

File: src/main/java/seedu/address/model/trip/Trip.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Trip(Driver driver, TripDay tripDay, TripTime tripTime, Set<Passenger> passengers, Set<Tag> tags) {
+        requireAllNonNull(driver, tripDay, tripTime);
+
+        // ensures a trip cannot be created with no passengers
+        if (passengers.isEmpty()) {
+            throw new NullPointerException();
```
Comment (JoelHo) [inline src/main/java/seedu/address/model/trip/Trip.java]: NullPointerException is probably the wrong exception to throw

**Review by JoelHo — CHANGES_REQUESTED** [review]
Some changes required in Trip.java

File: src/main/java/seedu/address/model/trip/Trip.java (hunk 2 above)
Comment (ZechariahTan) [reply src/main/java/seedu/address/model/trip/Trip.java]: You're right. I was intending to throw the same exception that the requireAllNotNull() check in the line above it does but an illegal argument exception would be more fitting.

File: src/main/java/seedu/address/model/trip/Trip.java (hunk 1 above)
Comment (ZechariahTan) [reply src/main/java/seedu/address/model/trip/Trip.java]: Noted thanks for the heads up.

**Review by JoelHo — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/pool/Pool.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Pool)) {
+            return false;
+        }
+
+        Pool otherPool = (Pool) other;
+        return otherPool.getDriver().equals(getDriver())
```
Comment (lyueyang) [inline src/main/java/seedu/address/model/pool/Pool.java]: could we actually just change this to isSameTrip to keep the code shorter?

**Review by lyueyang — APPROVED** [review]
Issue highlighted in previous comment has been resolved, LGTM :)

**Pull request merged**